modules/core/28-wasm/keeper/keeper.go

```diff
@@ -0,0 +1,91 @@
+package keeper
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"strings"
+
+	wasm "github.com/CosmWasm/wasmvm"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/ibc-go/modules/core/28-wasm/types"
+)
+
+// WasmVM initialized by wasm keeper
+var WasmVM *wasm.VM
+
+// VMConfig represents Wasm virtual machine settings
+type VMConfig struct {
+	DataDir           string
+	SupportedFeatures []string
+	MemoryLimitMb     uint32
+	PrintDebug        bool
+	CacheSizeMb       uint32
+}
+
+// Keeper will have a reference to Wasmer with it's own data directory.
+type Keeper struct {
+	storeKey      sdk.StoreKey
+	cdc           codec.BinaryCodec
+	wasmValidator *WasmValidator
+}
+
+func NewKeeper(cdc codec.BinaryCodec, key sdk.StoreKey, vmConfig *VMConfig, validationConfig *ValidationConfig) Keeper {
+	supportedFeatures := strings.Join(vmConfig.SupportedFeatures, ",")
+
+	vm, err := wasm.NewVM(vmConfig.DataDir, supportedFeatures, vmConfig.MemoryLimitMb, vmConfig.PrintDebug, vmConfig.CacheSizeMb)
+	if err != nil {
+		panic(err)
+	}
+
+	wasmValidator, err := NewWasmValidator(validationConfig, func() (*wasm.VM, error) {
+		return wasm.NewVM(vmConfig.DataDir, supportedFeatures, vmConfig.MemoryLimitMb, vmConfig.PrintDebug, vmConfig.CacheSizeMb)
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	WasmVM = vm
```

Comment:
Calling NewKeeper from concurrent goroutines would yield a data race and memory model violation. Does IBC specify or somehow guarantee that packages under `modules` will never be accessed concurrently?